pkg/storage/unified/apistore/restore.go

```diff
@@ -0,0 +1,102 @@
+package apistore
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/registry/generic"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	"github.com/grafana/grafana/pkg/storage/unified/resource"
+)
+
+type RestoreConnector interface {
+	rest.Storage
+	rest.Connecter
+	rest.StorageMetadata
+}
+
+func NewRestoreConnector(unified resource.ResourceClient, gr schema.GroupResource, opts generic.RESTOptions) RestoreConnector {
+	return &restoreREST{
+		unified: unified,
+		gr:      gr,
+		opts:    opts,
+	}
+}
+
+type restoreREST struct {
+	unified resource.ResourceClient
+	gr      schema.GroupResource
+	opts    generic.RESTOptions
+}
+
+func (r *restoreREST) New() runtime.Object {
+	return &metav1.PartialObjectMetadataList{}
+}
+
+func (r *restoreREST) Destroy() {
+}
+
+func (r *restoreREST) ConnectMethods() []string {
+	return []string{"POST"}
+}
+
+func (r *restoreREST) ProducesMIMETypes(verb string) []string {
+	return nil
+}
+
+func (r *restoreREST) ProducesObject(verb string) interface{} {
+	return &metav1.PartialObjectMetadataList{}
+}
+
+func (r *restoreREST) NewConnectOptions() (runtime.Object, bool, string) {
+	return nil, false, ""
+}
+
+func (r *restoreREST) Connect(ctx context.Context, uid string, opts runtime.Object, responder rest.Responder) (http.Handler, error) {
+	info, err := NamespaceInfoFrom(ctx, true)
+	if err != nil {
+		return nil, err
+	}
+
+	key := &resource.ResourceKey{
+		Namespace: info.Value,
+		Group:     r.gr.Group,
+		Resource:  r.gr.Resource,
+		Name:      uid,
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		query := req.URL.Query()
+		res := query.Get("resourceVersion")
```

Comment:
I wonder if it'd make more sense to use parametrized body approach instead, that way we can generate typed request structs.

Kubernetes uses this for e.g. `pods/proxy` – https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L5716-L5722:
```json
{
  "name": "pods/proxy",
  "singularName": "",
  "namespaced": true,
  "kind": "PodProxyOptions",
  "verbs": [
    "create",
    "delete",
    "get",
    "patch",
    "update"
  ]
}
```

Reply:
Updated :)